src/lightning/fabric/connector.py

```diff
@@ -377,8 +377,8 @@ def _choose_and_init_cluster_environment(self) -> ClusterEnvironment:
         if isinstance(self._cluster_environment_flag, ClusterEnvironment):
             return self._cluster_environment_flag
         for env_type in (
-            SLURMEnvironment,
             TorchElasticEnvironment,
+            SLURMEnvironment,
```

Comment:
would it help to document that the order of the above 2 is important so that someone won't "fix" this inadvertently?

Reply:
I added a comment there if that's ok. But yeah if the order gets reversed, the test that I added would fail.